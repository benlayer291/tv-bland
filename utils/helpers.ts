export const removeProtocolFromUrl = (url: string): string =>
  url.replace(/^https?:\/\//i, '//')

export const removeHtmlTagsFromString = (htmlString: string): string =>
  htmlString.replace(/(<([^>]+)>)/gi, '')

export const checkDeviceIsMobile = (): boolean => {
  return (
    typeof navigator !== 'undefined' &&
    /Android|Mobi/i.test(navigator.userAgent)
  )
}
