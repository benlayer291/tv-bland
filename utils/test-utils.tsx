import { render, RenderOptions } from '@testing-library/react'

type Props = {
  children: React.ReactElement
}

const Providers: React.FunctionComponent<Props> = ({ children }: Props) => {
  return <>{children}</>
}

const customRender = (
  ui: React.ReactElement,
  options: Omit<RenderOptions, 'queries'> = {}
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
