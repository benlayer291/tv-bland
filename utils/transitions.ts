type Transtion = {
  duration: number
  ease: number[]
}

export const transition: Transtion = {
  duration: 5,
  ease: [0.43, 0.13, 0.23, 0.96],
}
