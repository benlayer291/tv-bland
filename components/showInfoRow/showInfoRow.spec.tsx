import React from 'react'
import { render, RenderResult } from '../../utils/test-utils'

import ShowInfoRow from './showInfoRow'

const mockRow = {
  left: 'Left',
  right: 'Right',
}

describe('<ShowInfoRow />', () => {
  let el: RenderResult

  it('should render', () => {
    el = render(<ShowInfoRow row={{ ...mockRow, image: 'imageSrc' }} />)

    expect(el.container).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    el = render(<ShowInfoRow row={{ ...mockRow, image: 'imageSrc' }} />)

    expect(el.container).toMatchSnapshot()
  })

  describe('given no image is passed', () => {
    it('should render', () => {
      el = render(<ShowInfoRow row={mockRow} />)

      expect(el.container).toBeInTheDocument()
    })

    it('should match snapshot', () => {
      el = render(<ShowInfoRow row={mockRow} />)

      expect(el.container).toMatchSnapshot()
    })

    it('it should not contain an image', () => {
      el = render(<ShowInfoRow row={mockRow} />)

      expect(el.queryByRole('image')).not.toBeInTheDocument()
    })
  })
})
