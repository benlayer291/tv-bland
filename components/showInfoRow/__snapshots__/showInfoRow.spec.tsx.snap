// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowInfoRow /> given no image is passed should match snapshot 1`] = `
<div>
  <div
    class="ShowInfoRow"
  >
    <div
      class="ShowInfoText ShowInfoRow__item"
    >
      <span
        class="ShowInfoText__left"
      >
        <span>
          Left
        </span>
      </span>
      <span
        class="ShowInfoText__right"
      >
        <span>
          Right
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`<ShowInfoRow /> should match snapshot 1`] = `
<div>
  <div
    class="ShowInfoRow ShowInfoRow--center ShowInfoRow--full"
  >
    <div
      class="ShowInfoImage ShowInfoRow__item--noFlex"
    >
      <div
        class="BaseRevealImage"
        data-testid="base-reveal-image"
      >
        <img
          alt="Portrait of Left"
          class="BaseRevealImage__image"
          height="295"
          loading="lazy"
          src="imageSrc"
          width="210"
        />
        <img
          alt="Portrait of Left"
          class="BaseRevealImage__filter"
          height="295"
          loading="lazy"
          src="imageSrc"
          width="210"
        />
      </div>
    </div>
    <div
      class="ShowInfoText ShowInfoRow__item ShowInfoText--reverse"
    >
      <span
        class="ShowInfoText__left"
      >
        <span>
          Left
        </span>
      </span>
      <span
        class="ShowInfoText__right"
      >
        <span>
          Right
        </span>
      </span>
    </div>
  </div>
</div>
`;
